src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -109,11 +109,36 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
             override fun visitPatTupleStruct(o: RsPatTupleStruct) = checkRsPatTupleStruct(rsHolder, o)
             override fun visitPatTup(o: RsPatTup) = checkRsPatTup(rsHolder, o)
             override fun visitStructLiteralField(o: RsStructLiteralField) = checkReferenceIsPublic(o, o, rsHolder)
+            override fun visitMetaItem(o: RsMetaItem) = checkMetaItem(rsHolder, o)
         }
 
         element.accept(visitor)
     }
 
+    private fun checkMetaItem(holder: RsAnnotationHolder, metaItem: RsMetaItem) {
+        val args = metaItem.metaItemArgs
+        val name = metaItem.name
+        if (args != null && name in listOf("cfg", "cfg_attr")) {
+            val item = args.metaItemList.getOrNull(0) ?: return
+            checkCfgPredicate(holder, item)
+        }
+    }
+
+    private fun checkCfgPredicate(holder: RsAnnotationHolder, item: RsMetaItem) {
+        val itemName = item.name ?: return
+        val args = item.metaItemArgs
+        if (args != null) {
+            when (itemName) {
+                "all", "any" -> args.metaItemList.forEach { checkCfgPredicate(holder, it) }
+                "not" -> {
+                    val parameter = args.metaItemList.getOrNull(0) ?: return
+                    checkCfgPredicate(holder, parameter)
+                }
+                else -> RsDiagnostic.UnknownCfgPredicate(item, itemName).addToHolder(holder)
```

Comment:
Not a call to action: don't like that compiler highlights the whole attribute instead of the identifier only

Reply:
I'd prefer an item name

Reply:
This mirrors the behavior of `rustc`:
![image](https://user-images.githubusercontent.com/4539057/114842465-276f4600-9dd9-11eb-9928-f3704ce74fca.png)
But if you prefer to highlight only the attribute name, I will change it. I suppose that it would be more readable for the user.

Reply:
We've discussed it with @vlad20012 and decided that it's ok to have a narrower highlighting range in comparison with the compiler